From 765afda8e8fb6b2a8a013a3b952d731e6afb887c Mon Sep 17 00:00:00 2001
From: Leonardo Martins Martins <dev.lmmrtns@gmail.com>
Date: Tue, 21 Jul 2026 15:37:46 -0300
Subject: [PATCH 07598/15333] staging: rtl8723bs: remove redundant goto in
 rtw_free_xmitframe()

Remove redundant goto statement and return _SUCCESS directly.

Signed-off-by: Leonardo Martins Martins <dev.lmmrtns@gmail.com>
Link: https://patch.msgid.link/20260721183746.80069-1-dev.lmmrtns@gmail.com
Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
---
 drivers/staging/rtl8723bs/core/rtw_xmit.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/drivers/staging/rtl8723bs/core/rtw_xmit.c b/drivers/staging/rtl8723bs/core/rtw_xmit.c
index 16e6c9567224..f11fc285ee3a 100644
--- a/drivers/staging/rtl8723bs/core/rtw_xmit.c
+++ b/drivers/staging/rtl8723bs/core/rtw_xmit.c
@@ -1740,7 +1740,7 @@ s32 rtw_free_xmitframe(struct xmit_priv *pxmitpriv, struct xmit_frame *pxmitfram
 	struct sk_buff *pndis_pkt = NULL;
 
 	if (!pxmitframe)
-		goto exit;
+		return _SUCCESS;
 
 	if (pxmitframe->pkt) {
 		pndis_pkt = pxmitframe->pkt;
@@ -1773,7 +1773,6 @@ s32 rtw_free_xmitframe(struct xmit_priv *pxmitpriv, struct xmit_frame *pxmitfram
 	if (pndis_pkt)
 		rtw_os_pkt_complete(padapter, pndis_pkt);
 
-exit:
 	return _SUCCESS;
 }
 
-- 
2.55.0

